therapi-runtime-javadoc/src/main/java/com/github/therapi/runtimejavadoc/internal/JsonJavadocReader.java

```diff
@@ -0,0 +1,75 @@
+package com.github.therapi.runtimejavadoc.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+import com.github.therapi.runtimejavadoc.ClassJavadoc;
+import com.github.therapi.runtimejavadoc.FieldJavadoc;
+import com.github.therapi.runtimejavadoc.MethodJavadoc;
+
+import static com.github.therapi.runtimejavadoc.internal.RuntimeJavadocHelper.elementDocFieldName;
+import static com.github.therapi.runtimejavadoc.internal.RuntimeJavadocHelper.elementNameFieldName;
+import static com.github.therapi.runtimejavadoc.internal.RuntimeJavadocHelper.enumConstantsFieldName;
+import static com.github.therapi.runtimejavadoc.internal.RuntimeJavadocHelper.fieldsFieldName;
+import static com.github.therapi.runtimejavadoc.internal.RuntimeJavadocHelper.methodsFieldName;
+import static com.github.therapi.runtimejavadoc.internal.RuntimeJavadocHelper.paramTypesFieldName;
+
+public class JsonJavadocReader {
+
+    public static Optional<ClassJavadoc> readClassJavadoc(String qualifiedClassName, JsonObject json) {
+        String className = qualifiedClassName.replace("$", ".");
+        List<FieldJavadoc> fields = readFieldDocs(json.get(fieldsFieldName()));
+        List<FieldJavadoc> enumConstants = readFieldDocs(json.get(enumConstantsFieldName()));
+        List<MethodJavadoc> methods = readMethodDocs(json.get(methodsFieldName()));
+        String classJavadocString = json.getString(elementDocFieldName(), null);
+        ClassJavadoc classJavadoc = JavadocParser.parseClassJavadoc(className, classJavadocString, fields,
+                enumConstants, methods);
+        return Optional.of(classJavadoc);
+    }
+
+    private static List<FieldJavadoc> readFieldDocs(JsonValue fieldsValue) {
+        JsonArray fieldsArray = fieldsValue.asArray();
+        List<FieldJavadoc> fields = new ArrayList<>(fieldsArray.size());
+        for (JsonValue fieldValue : fieldsArray) {
+            fields.add(readFieldDoc(fieldValue));
+        }
+        return fields;
+    }
+
+    private static FieldJavadoc readFieldDoc(JsonValue fieldValue) {
+        JsonObject field = fieldValue.asObject();
```

Comment:
JSON resources generated by previous versions will not have the fields or enums properties.
What do you think about making it backwards compatible by checking for null `fieldValue` and returning empty list in that case?

Reply:
It might be wise indeed to make this safer, as this could indeed happen if someone uses the old annotation processor with the new runtime-javadoc dependency, or has a subproject using an older version of the processor.
I'll try to have a look today.

Reply:
Since it was not too costly in terms of readability and code, I made the change to make it backwards compatible regarding the absence of fields/enum constants in the JSON resource.

That being said, the same kind of issues would have occurred when switching from classes to resources. I guess at some point the users have to align their versions if they expect proper generation and retrieval of the doc.